crates/bifrost/src/error.rs

```diff
@@ -42,6 +43,8 @@ pub enum Error {
     AdminError(#[from] AdminError),
     #[error(transparent)]
     MetadataStoreError(#[from] restate_core::metadata_store::ReadWriteError),
+    #[error("partition not found: {0}")]
+    PartitionNotFound(#[from] PartitionTableError),
```

Comment:
I would create a new `AppendError` that has this and `Error` as a variant to not infect `Error` with the partition lookup error.